### PR TITLE
STOP WATCHing

### PR DESCRIPTION
**Pull request by abcdesign1**
- [DEMO LINK](https://vadym4che.github.io/layout_stop-watch/)

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1 +1,22 @@
...
+    + new Date().getMinutes() * 60
+    + new Date().getSeconds()
+  ) + 's';
+
+  document.querySelector('.watch__seconds-second').style = seconds;
+
+  document.querySelector('.watch__minutes-minute').style = minutes;
+
+  document.querySelector('.watch__hours-hour').style = hours;
+};
```
Comment (sergii-nosachenko) [inline src/main.js:22]: This task is not to use javascript to create the clock. It's about "stopwatch using CSS animation".
So your solution doesn't satisfy task requirements.

File: src/styles/main.scss [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+$c-minutes: hsl(131, 75%, 50%);
+$c-min-axis: hsl(131, 75%, 50%);
+$c-min-glow: hsl(131, 75%, 50%);
+
+$c-hours: hsl(131, 50%, 45%);
+$c-hour-glow: hsl(131, 50%, 45%);
+$c-hour-axis: hsl(131, 50%, 45%);
+
+$c-lines: hsla(130, 50%, 50%, 0.68);
+$c-circles: rgba(68, 196, 68, 0.32);
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:14]: Use one color model everywhere.

File: src/styles/main.scss [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+$size-glow: 0.5;
+$size-glow-blur: 0.1;
+
+$size-main: 75;
+$size-circle: 9;
+$size-pad: 70;
+$size-second: 62;
+$size-minute: 63.5;
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:23]: Whi not to sent units here?

File: src/styles/main.scss [hunk 4]
```diff
@@ -0,0 +1,284 @@
+$c-seconds: rgb(0, 255, 45);
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:1]: Create separate file for variables

File: src/styles/main.scss [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+$size-minute: 63.5;
+
+$size-main-ls: #{$size-main}vh;
+$size-main-pt: #{$size-main}vw;
+
+$size-br-radius: 1vh;
+
+* {
+  box-sizing: border-box;
+}
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:32]: Reset file

File: src/styles/main.scss [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+  background: conic-gradient(
+    rgba(255, 255, 255, 0.8),
+    rgba(0, 0, 0, 0.2),
+    rgba(255, 255, 255, 0.8)
+  );
+  height: 100vh;
+  margin: 0;
+  display: flex;
+  justify-content: center;
+}
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:44]: Create block for this

File: src/styles/main.scss [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+* {
+  box-sizing: border-box;
+}
+
+body {
+  background: conic-gradient(
+    rgba(255, 255, 255, 0.8),
+    rgba(0, 0, 0, 0.2),
+    rgba(255, 255, 255, 0.8)
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:38]: Why these are not set in variables?

File: src/styles/main.scss [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+  margin: auto;
+  @media (orientation: landscape) {
+    width: $size-main-ls;
+    height: $size-main-ls;
+  }
+  @media (orientation: portrait) {
+    width: $size-main-pt;
+    height: $size-main-pt;
+  }
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:60]: Combine these queries with same below

File: src/styles/main.scss [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+.watch {
+
+  & > * {
+    opacity: 0.68;
+  }
+
+  margin: auto;
+  @media (orientation: landscape) {
+    width: $size-main-ls;
+    height: $size-main-ls;
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:55]: Create mixin

File: src/styles/main.scss [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+    width: $size-main-pt;
+    height: $size-main-pt;
+  }
+  position: relative;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  border-radius: 50%;
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:67]: Add to `circle` mixin (with `$size` argument)

File: src/styles/main.scss [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+  &__seconds {
+
+    @media (orientation: landscape) {
+      width: #{$size-second}vh;
+      height: #{$size-second}vh;
+    }
+    @media (orientation: portrait) {
+      width: #{$size-second}vw;
+      height: #{$size-second}vw;
+    }
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:114]: Set media queries after all element props

File: src/styles/main.scss [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+      0% {
+        transform: rotate(0);
+      }
+      @for $step from 1 through 60 {
+
+        #{$step * 10/6}% {
+          transform: rotate(#{($step) * 6}deg);
+        }
+      }
+    }
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:128]: Create separate file for animation keyframes

File: src/styles/main.scss [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:189]: Looks like the same as `seconding`

File: src/styles/main.scss [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:245]: Reuse one animation

File: src/styles/main.scss [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+        left: 45.5%;
+        width: 9%;
+        height: 9%;
+        border-radius: 50%;
+        background: $c-hour-axis;
+        box-shadow: 0 0 #{$size-glow}vw #{$size-glow-blur}vw $c-hour-glow;
+      }
+    }
+  }
+}
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:284]: A lot of repeated code for different stopwatch elements. Try to write as little code as possible. Create mixins, placeholders, functions, loops, etc. to achieve this.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]
I like your idea to create something extraordinary 👍, but first of all, you need to stay within the task requirements.

File: src/main.js (hunk 1 above)
Comment (vadym4che) [reply src/main.js:22]: removed

File: src/styles/main.scss (hunk 2 above)
Comment (vadym4che) [reply src/styles/main.scss:14]: unified to one type

File: src/styles/main.scss (hunk 3 above)
Comment (vadym4che) [reply src/styles/main.scss:23]: fixed
(I've brought my mind back to units `vmin` only after specific lesson wich went after I've been making this...)))

File: src/styles/main.scss (hunk 4 above)
Comment (vadym4che) [reply src/styles/main.scss:1]: separated

File: src/styles/main.scss (hunk 5 above)
Comment (vadym4che) [reply src/styles/main.scss:32]: initiated specially..

File: src/styles/main.scss (hunk 11 above)
Comment (vadym4che) [reply src/styles/main.scss:114]: as I don't use this weird approach - I don't really need those states in this case anymore

File: src/styles/main.scss (hunk 12 above)
Comment (vadym4che) [reply src/styles/main.scss:128]: done

File: src/styles/main.scss (hunk 13 above)
Comment (vadym4che) [reply src/styles/main.scss:189]: that it is!)

File: src/styles/main.scss (hunk 14 above)
Comment (vadym4che) [reply src/styles/main.scss:245]: got it

File: src/styles/main.scss (hunk 15 above)
Comment (vadym4che) [reply src/styles/main.scss:284]: trying to be smarter and keep it more readable and reusable
thank you

File: src/styles/main.scss (hunk 7 above)
Comment (illiarerun) [reply src/styles/main.scss:15]: You haven't fixed it, either store entire gradient in variable, or colors by themselves

File: src/styles/main.scss (hunk 6 above)
Comment (illiarerun) [reply src/styles/main.scss:21]: agree, don't use tag selectors for styling

File: src/styles/main.scss (hunk 10 above)
Comment (illiarerun) [reply src/styles/main.scss:39]: fix this comment

File: src/styles/main.scss [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+    position: absolute;
+    z-index: 2;
+
+    &-minute {
+
+      @extend %abs100;
+
+      animation-name: stepOn60;
+      animation-duration: 3600s;
+      animation-iteration-count: infinite;
```
Comment (illiarerun) [inline src/styles/main.scss:123]: create mixin for it with duration param

File: src/styles/main.scss [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+        box-shadow: 0 0 $size-glow $size-glow-blur $c-sec-glow;
+      }
+
+      &::after {
+        content: "";
+        position: absolute;
+        top: 48%;
+        left: 48%;
+        width: 4%;
+        height: 4%;
```
Comment (illiarerun) [inline src/styles/main.scss:101]: where does these values come from?

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/styles/main.scss (hunk 16 above)
Comment (vadym4che) [reply src/styles/main.scss:123]: fixed (and even separated that mixin to file)

File: src/styles/main.scss (hunk 17 above)
Comment (vadym4che) [reply src/styles/main.scss:89]: its a dot that symbolize axis of watch hand
to place it centered - I used 50% - 1/2 of its own height
I decided to not put this in formula to code not to be overcomplicated

File: src/index.html [hunk 18]
```diff
@@ -1,14 +1,57 @@
...
+          <use href="#lines" transform="rotate(60)"></use>
+          <use href="#lines" transform="rotate(66)"></use>
+          <use href="#lines" transform="rotate(72)"></use>
+          <use href="#lines" transform="rotate(78)"></use>
+          <use href="#lines" transform="rotate(84)"></use>
+        </g>
+      </svg>
+      <div class="watch__hours">
+        <div class="watch__hours-hour"></div>
+      </div>
```
Comment (sergii-nosachenko) [inline src/index.html:47]: It doesn't visible at all. Check it.

File: src/styles/_animation-stepOn60.scss [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+
+  0% {
+    transform: rotate(0);
+  }
+  @for $step from 1 through 60 {
+
+    #{$step * 10 / 6 * 1%} {
+      transform: rotate(#{$step * 6deg});
+    }
+  }
```
Comment (sergii-nosachenko) [inline src/styles/_animation-stepOn60.scss:11]:
```suggestion
  to {
    transform: rotate(1turn);
  }
```
Over-thinking.

File: src/styles/main.scss [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+
+      &::before {
+        border-radius: $size-br-radius;
+
+        content: "";
+        position: absolute;
+        top: 0;
+        left: 49%;
+        width: 2%;
+        height: 50%;
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:78]: It seems that all these groups of values can be calculated using one function or mixin. You've hardcoded a lot of data, it's bad practice.

File: src/styles/main.scss [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+  margin: 0;
+
+  @extend %flex-center;
+}
+
+.watch {
+
+  & > * {
+    opacity: 0.68;
+  }
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:19]: Bad idea. Add opacity directly to elements or blocks.

File: src/styles/main.scss [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+@import "./extends";
+@import "./mixins";
+
+.page {
+  background: $c-bg-gradient;
+  height: 100vh;
+  margin: 0;
+
+  @extend %flex-center;
+}
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:13]: Move to separate file.

File: src/styles/main.scss [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+
+.watch {
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:15]: Move to separate file

File: src/styles/main.scss (hunk 10 above)
Comment (sergii-nosachenko) [reply src/styles/main.scss:30]: Fix this too

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/styles/_animation-stepOn60.scss (hunk 19 above)
Comment (vadym4che) [reply src/styles/_animation-stepOn60.scss:11]: commited

File: src/styles/main.scss (hunk 21 above)
Comment (vadym4che) [reply src/styles/main.scss:19]: removed with respect to your advise

File: src/styles/main.scss (hunk 23 above)
Comment (vadym4che) [reply src/styles/main.scss:15]: moved to `page.scss` dedicated to page body general styling rules

Comment (vadym4che) [reply src/styles/main.scss:15]: optimised

File: src/styles/main.scss (hunk 20 above)
Comment (vadym4che) [reply src/styles/main.scss:78]: created 2 mixins for purpose to draw hands and their axis dots more mathematically effective

File: src/index.html (hunk 18 above)
Comment (vadym4che) [reply src/index.html:47]: removed because of  its unnecessarity

**Comment by vadym4che** [comment]
fixed and optimized

File: src/styles/main.scss [hunk 24]
```diff
@@ -3,29 +3,16 @@
...
-.page {
-  background: $c-bg-gradient;
-  height: 100vh;
-  margin: 0;
-
-  @extend %flex-center;
-}
+@import "./page";
 
 .watch {
```
Comment (sergii-nosachenko) [inline src/styles/main.scss:8]: `.watch` must be a separate file too

**Review by sergii-nosachenko — APPROVED** [review]

**Pull request closed without merging**